freqtrade/persistence/models.py

```diff
@@ -531,9 +531,10 @@ def update_fee(self, fee_cost: float, fee_currency: Optional[str], fee_rate: Opt
             self.fee_open_cost = fee_cost
             self.fee_open_currency = fee_currency
             if fee_rate is not None:
+                current_rate_ratio = self.fee_close / self.fee_open
```

Comment:
you cannot safely rely on this method being called only once (There are several scenarios where it'll be called multiple times, and also reach this line) per trade.

If this is called 2 or 3 times - then you'll encounter "disappearing" fees - as the close-fee will each time be updated lower - as the gap widens.